### PR TITLE
Add DaskDeviceQuantileDMatrix demo.

### LINKED ISSUE #6153: Create a demo for DaskDeviceQuantileDMatrix
DaskDeviceQuantileDMatrix is a nice feature to ingest GPU Dask partitions without making any copy. We should have a demo.

I'd like to get the word out about the native Dask API of XGBoost, and DaskDeviceQuantileDMatrix is one of its best features.

### PR DESCRIPTION
**Pull request by trivialfis**
Closes #6153

### CODE REVIEW

**Review by hcho3 — APPROVED** [review]

**Review by hcho3 — CHANGES_REQUESTED** [review]
In the commit message, please spell out `DaskDeviceQuantileDMatrix`. I was left wondering what DDQDM meant.

File: demo/dask/gpu_training.py [hunk 1]
```diff
@@ -31,15 +28,56 @@ def main(client):
...
         with Client(cluster) as client:
-            main(client)
+            # generate some random data for demonstration
+            m = 100000
+            n = 100
+            X = da.random.random(size=(m, n), chunks=100)
+            y = da.random.random(size=(m, ), chunks=100)
+
+            if args.ddqdm == 1:
+                print('Using QQDQM')
```
Comment (hcho3) [inline demo/dask/gpu_training.py:79]:
```suggestion
                print('Using DaskDeviceQuantileDMatrix')
```

**Review by hcho3 — CHANGES_REQUESTED** [review]

**Comment by trivialfis** [comment]
Done.

**Review by hcho3 — APPROVED** [review]

**Comment by hcho3** [comment]
Thanks a lot! I will point to this demo when I evangelize about XGBoost.Dask.

**Pull request merged**